### PR TITLE
Added service and controller for Surf

### PR DESCRIPTION
**Pull request by eiriksanderfjeld**
Added both controller and service based on Exploration. #154

### CODE REVIEW

**Review by petterwildhagen — CHANGES_REQUESTED** [review]
Fix codacy errors and take a look at Eric PR (soon to be merged to main) for splitting up db context for unit tests.

File: backend/api/Dtos/SurfDto.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        public Maturity Maturity { get; set; }
+        public double InfieldPipelineSystemLength { get; set; }
+        public double UmbilicalSystemLength { get; set; }
+        public ArtificialLift ArtificialLift { get; set; }
+        public int RiserCount { get; set; }
+        public int TemplateCount { get; set; }
+        public ProductionFlowline ProductionFlowline { get; set; }
+    }
+
+    public enum ProductionFlowlineDto
```
Comment (ericnorway) [inline backend/api/Dtos/SurfDto.cs:22]: I think this can be removed.

File: backend/api/Services/SurfService.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                throw new Exception(); //TODO FIX EXCEPTION
+            }
+        }
+
+        public Surf CreateSurf(Surf surf)
+        {
+            if (_context.Surfs != null)
+            {
+                var result = _context.Surfs.Add(surf);
```
Comment (ericnorway) [inline backend/api/Services/SurfService.cs:55]: I think the new surf needs to be added to the project's surf collection.

File: backend/api/Services/SurfService.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                throw new Exception(); //TODO FIX EXCEPTION
+            }
+        }
+
+        public bool DeleteSurf(Surf surf)
+        {
+            if (_context.Surfs != null)
+            {
+                var result = _context.Surfs.Remove(surf);
```
Comment (ericnorway) [inline backend/api/Services/SurfService.cs:69]: And removed from the project's surfs collection.

**Review by ericnorway — DISMISSED** [review]
approved

**Pull request closed without merging**